Modules/TPC/src/PadCalibrationCheck2D.cxx

```diff
@@ -0,0 +1,144 @@
+{
+
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Null;
+  if (mo->getName() == "c_ROCs_Pedestal_2D") {
+    result = Quality::Good;
+    auto* canv = dynamic_cast<TCanvas*>(mo->getObject());
+    // Check all histograms in the canvas
+    for (int tpads = 1; tpads <= 72; tpads++) {
+      auto padName = fmt::format("c_ROCs_Pedestal_2D_{:d}", tpads);
+      auto histName = fmt::format("h_Pedestals_ROC_{:02d}", tpads - 1);
+      TPad* pad = (TPad*)canv->GetListOfPrimitives()->FindObject(padName.data());
+      TH2F* h = (TH2F*)pad->GetListOfPrimitives()->FindObject(histName.data());
+      std::string titleh = h->GetTitle();
+      bool iroc = false;
+      bool oroc = false;
+      //check if we are dealing with IROC or OROC
+      if (titleh.find("IROC") != std::string::npos)
+        iroc = true;
+      if (titleh.find("OROC") != std::string::npos)
+        oroc = true;
+      if (!(iroc || oroc))
+        return Quality::Null;
+
+      int NX = h->GetNbinsX();
+      int NY = h->GetNbinsY();
+
+      int totalPads = 0;
+      if (iroc)
+        totalPads = 5280;
+      if (oroc)
+        totalPads = 9280;
+
+      // Check how many of the pads are non zero
+      int sum = 0;
+      for (int i = 1; i <= NX; i++) {
+        for (int j = 1; j <= NY; j++) {
+          int val = h->GetBinContent(i, j);
+          if (val > 0) {
+            sum += 1;
+          }
+        }
+      }
+
+      // Check how many are off
+      float mediumLimit = 0.7;
+      float badLimit = 0.4;
+      if (sum > badLimit * totalPads && sum < mediumLimit * totalPads) {
+        if (result == Quality::Good)
+          result = Quality::Medium;
+        badSectorsName.push_back(titleh);
+        badSectorsQuality.push_back(Quality::Medium);
+      } else if (sum < badLimit * totalPads) {
+        result = Quality::Bad;
+        badSectorsName.push_back(titleh);
+        badSectorsQuality.push_back(Quality::Bad);
+      } else {
+        badSectorsName.push_back(titleh);
+        badSectorsQuality.push_back(Quality::Good);
+      }
+    }
+  }
+  return result;
+}
+
+std::string PadCalibrationCheck2D::getAcceptedType() { return "TCanvas"; }
+
+void PadCalibrationCheck2D::beautify(std::shared_ptr<MonitorObject> mo, Quality)
+{
+
+  auto* canv = dynamic_cast<TCanvas*>(mo->getObject());
+  // Loop over all histograms and beautify accordingly to specific quality object
+  for (int tpads = 1; tpads <= 72; tpads++) {
+    auto padName = fmt::format("c_ROCs_Pedestal_2D_{:d}", tpads);
+    auto histName = fmt::format("h_Pedestals_ROC_{:02d}", tpads - 1);
+    TPad* pad = (TPad*)canv->GetListOfPrimitives()->FindObject(padName.data());
+    pad->cd();
+    TH2F* h = (TH2F*)pad->GetListOfPrimitives()->FindObject(histName.data());
+    std::string titleh = h->GetTitle();
+    std::vector<std::string>::iterator it = std::find(badSectorsName.begin(), badSectorsName.end(), titleh);
+    if (it != badSectorsName.end()) {
```

Comment:
Change logic to make clear that the reader does not have to continue reading the code:
```suggestion
    if (it == badSectorsName.end()) {
      continue
    }
```